internal/provider/base_eda_datasource.go

```diff
@@ -0,0 +1,198 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/datasourcevalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	tfpath "github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource                     = &BaseEdaDataSource{}
+	_ datasource.DataSourceWithConfigure        = &BaseEdaDataSource{}
+	_ datasource.DataSourceWithConfigValidators = &BaseEdaDataSource{}
+	_ datasource.DataSourceWithValidateConfig   = &BaseEdaDataSource{}
+)
+
+// NewBaseEdaDataSource creates a new `BaseEdaDataSource`.
+func NewBaseEdaDataSource(client ProviderHTTPClient, stringDescriptions StringDescriptions) *BaseEdaDataSource {
+	return &BaseEdaDataSource{
+		client:             client,
+		StringDescriptions: stringDescriptions,
+	}
+}
+
+// Metadata returns the data source type name composing it from the provider type name and the
+// entity slug string passed in the constructor.
+func (d *BaseEdaDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, d.MetadataEntitySlug)
+	fmt.Println(resp.TypeName)
+}
+
+// Schema defines the schema fields for the data source.
+func (d *BaseEdaDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+			"url": schema.StringAttribute{
+				Computed: true,
+			},
+			"id": schema.Int64Attribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+// ConfigValidators validates configuration in a declarative way.
+func (d *BaseEdaDataSource) ConfigValidators(_ context.Context) []datasource.ConfigValidator {
+	// You have at least an name
+	return []datasource.ConfigValidator{
+		datasourcevalidator.Any(
+			datasourcevalidator.AtLeastOneOf(
+				tfpath.MatchRoot("name")),
+		),
+	}
+}
```

Comment:
Since `name` is already a required attribute in the schema, is this validator necessary? 

I think in general validators can be used for complex either/or relationships or to make sure strings match some pattern. Maybe we have these in other datasources, but I think if we're just making sure `name` exists, the `Required: true` on the Attribute handles it.